### ISSUE TITLE
Exception

### ISSUE THREAD

**Issue by egrimate** [jgraph/svg2xml#24]
Asked in google group here: https://groups.google.com/g/drawio/c/iwRh9zaH3A4
Any suggestion on cause and fix or workaround for this exception running sv2xml tool on this svg with default options:

`<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 32 32"><defs><style>.cls-1{fill:#fff;}.cls-2{fill:url(#linear-gradient);}.cls-3{mask:url(#mask);}.cls-4{fill:#161616;}</style><linearGradient id="linear-gradient" x1="2273.534" y1="-956.061" x2="2273.534" y2="-976.061" gradientTransform="matrix(1, 0, 0, -0.99, -2257.534, -940.352)" gradientUnits="userSpaceOnUse"><stop offset="0.05"/><stop offset="0.6" stop-opacity="0"/></linearGradient><mask id="mask" x="0" y="0" width="32" height="32" maskUnits="userSpaceOnUse"><path class="cls-1" d="M23.707,22.293,10,8.586V2H2v8H8.586L22.293,23.707ZM8,8H4V4H8Z"/><rect class="cls-2" x="11" y="6.101" width="10" height="19.799" transform="translate(38.627 16) rotate(135)"/><path class="cls-1" d="M30,30H22V22h8Zm-6-2h4V24H24Z"/></mask></defs><g id="Layer_2" data-name="Layer 2"><g id="Build_icon_here" data-name="Build icon here"><g class="cls-3"><rect class="cls-4" width="32" height="32"/></g><path class="cls-4" d="M7,22.142V13H5v9.142A3.992,3.992,0,1,0,9.858,27H19V25H9.858A4,4,0,0,0,7,22.142ZM6,28a2,2,0,1,1,2-2A2,2,0,0,1,6,28Z"/><path class="cls-4" d="M30,6a3.992,3.992,0,0,0-7.858-1H13V7h9.142A4,4,0,0,0,25,9.858V19h2V9.858A4,4,0,0,0,30,6ZM26,8a2,2,0,1,1,2-2A2,2,0,0,1,26,8Z"/></g></g></svg>`

Exception in thread "AWT-EventQueue-0" java.lang.StringIndexOutOfBoundsException: String index out of range: 0

	at java.lang.String.charAt(String.java:658)

	at com.mxgraph.svg2xml.Svg2Xml.getAvgGradientColor(Svg2Xml.java:1465)

	at com.mxgraph.svg2xml.Svg2Xml.parseFillDefs(Svg2Xml.java:1388)

	at com.mxgraph.svg2xml.Svg2Xml.parseFillDefs(Svg2Xml.java:1392)

	at com.mxgraph.svg2xml.Svg2Xml.parseFills(Svg2Xml.java:1305)

	at com.mxgraph.svg2xml.Svg2Xml.flattenSvg(Svg2Xml.java:1292)

	at com.mxgraph.svg2xml.Svg2Xml.<init>(Svg2Xml.java:317)

	at com.mxgraph.svg2xml.Svg2XmlGui.actionPerformed(Svg2XmlGui.java:253)

**Comment by tbouffard** [comment]
@egrimate can you provide the whole svg please? it is not possible to reproduce with the svg extract you have provided

**Comment by egrimate** [comment]
It was copied from the diagrams.net Google group, wasn't tested yet.

**Comment by tbouffard** [comment]
Ok thanks, the post on `diagrams.net Google group` is https://groups.google.com/g/drawio/c/iwRh9zaH3A4
It provides the full svg

**Comment by egrimate** [comment]
I've updated the issue. It now contains the link to the group and the SVG content is also shown in full.

**Comment by tbouffard** [comment]
Reproduced

The code causing the issue is the following: with the provided svg, the `colS` String is empty and inverting the if/else-if condition should fix the problem

```java
if (colS.charAt(0) == '#')
{
	col = Color.decode(colS);
}
else if (colS == null || colS.equals("")) 
{
	return null;
}
```
Notice that the parseStrokeDefs and parseFillDefs methods should then be updated to avoid NullPointerException caused by an attempt to add a `null` value in Hashtable. See https://github.com/process-analytics/mxgraph-svg2shape/pull/32 for a fix attempt.

With that fix, the generated stencil is: [svg2xml_issue_24.zip](https://github.com/jgraph/svg2xml/files/5780381/svg2xml_issue_24.zip)

I am not sure this is what is expected but at least it is generated 😸

**Comment by jaywarfield** [comment]
I will try out the generated stencil with the fix, thanks!

**Comment by jaywarfield** [comment]
I tried the svg2xml_issue_24.zip, looks close but I'm seeing issue 25 here too.
The icon in the xml should look like drawiotestinput.svg in the zip here.
Rendering the icon in the xml in draw.io (standard draw.io APIs) is the screenshot in drawiotestoutput1 in the zip.
The icon is completely black.  The word Zone next to the icon is text that is separate from the icon.
Like issue #25 there is the extra box in the xml (not in the original svg) that is hiding the icon itself.
The extra box is usually at the end of shape in the xml but this is the first time I've seen the extra box at the beginning of the shape in the xml.  
I tried deleting the extra box which are the statements <rect h"32" w="32" x="0" y="0"><fillstroke/>
The drawiotestoutput2 in the zip shows the icon rendered in drawio after deleting the extra box from the xml.  
The icon is almost complete after deleting the extra box but is missing the diagonal line as shown in drawiotestinput.svg.
It looks like you have a fix for the string exception which would be good to get into the svg2xml code when you think the fix is ready.
Thanks for taking a look at this!

[drawio_outcome_issue24.zip](https://github.com/jgraph/svg2xml/files/5804979/drawio_outcome_issue24.zip)
